### PR TITLE
Checkbox multi

### PR DESCRIPTION
**Pull request by mattheu**
Multi Checkbox field type - based on patch from @stompweb

Like checkbox, except allow more than one. Pass an array of options - like you do with radio. Saved as an array, with checked fields set. If none checked, no data is saved.
- The other frameworks have this.
- Need separate field to keep current behaviour of checkbox (set/unset) 

Question - do we need this over just using multiple single checkbox fields? Better UI?

### CODE REVIEW

**Comment by mikeselander** [comment]
This is now ready for review/merging after a rewrite.

File: example-functions.php [hunk 1]
```diff
@@ -99,7 +99,24 @@ function cmb_sample_metaboxes( array $meta_boxes ) {
...
 
-		/*
+		/**
+		 * Checkbox Multi Field.
+		 */
+		array(
+			'id' => 'field-7b',
+			'name' => 'Checkbox-Multi field',
+			'type' => 'checkbox_multi',
+			'default' => [ 'option1' ],
```
Comment (BronsonQuick) [inline example-functions.php]: Change to `array()`

File: tests/testCheckboxMultiField.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+class CheckboxMultiFieldTestCase extends WP_UnitTestCase {
+
+	function setUp() {
+		parent::setUp();
+
+		$args = array(
+			'post_author'  => 1,
+			'post_status'  => 'publish',
+			'post_content' => rand_str(),
```
Comment (BronsonQuick) [inline tests/testCheckboxMultiField.php]: 🔢 Random strings can create issues so you're better off changing it to a fixed string instead.

Comment (mikeselander) [inline tests/testCheckboxMultiField.php]: Wherever did you get that idea? ;)

**Pull request merged**